moveit_py/src/moveit/moveit_core/robot_model/joint_model_group.cpp

```diff
@@ -0,0 +1,61 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2022, Peter David Fagan
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of PickNik Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Peter David Fagan */
+
+#include "joint_model_group.h"
+
+namespace moveit_py
+{
+namespace bind_robot_model
+{
+void init_joint_model_group(py::module& m)
+{
+  py::class_<moveit::core::JointModelGroup>(m, "JointModelGroup",
+                                            R"(
+          Representation of a group of joints that are part of a robot model.
+          )")
+
+      .def_property("name", &moveit::core::JointModelGroup::getName, nullptr,
+                    R"(
+                    str: The name of the joint model group.
+                    )")
+
+      .def_property("joint_model_names", &moveit::core::JointModelGroup::getJointModelNames, nullptr,
+                    R"(
+                    list of str: The names of the joint models in the group.
```

Comment:
Is `list of str` how these docstrings should be written? I've seen `list[str]` in other places before, but whatever is correct per your standard

Reply:
This is a typo it should be list[str] as you have outlined above.

Reply:
[220864e](https://github.com/ros-planning/moveit2/pull/1546/commits/220864eae8ee0d518e040690100de0fe645bad95)